steps/step-camera/lib/angular/step.js

```diff
@@ -46,12 +46,12 @@ function initModule(cameraOptionsBuilder) {
         };
 
         self.displayImage = function(uri) {
+          self.model.data = {};
```

Comment:
only here I think we need to do that initialization `self.model.data = self.model.data || {};` to avoid overwriting other fields.